GeometryCore/src/Polygon.cpp

```diff
@@ -120,10 +121,88 @@ namespace geom_utils
     }
 
 
+    float Polygon::perpendicularDistance(const FPoint2D& point,const FPoint2D& lineStart, const FPoint2D& lineEnd)
+    {
+        FPoint2D dpoint (lineEnd - lineStart); 
+        //Normalise
+        float dist = distance(lineEnd, lineStart);
+        //Normalise
+        
+        if (dist > 0.0)
+        {
+            dpoint.x /= dist;
+            dpoint.y /= dist;
+        }
+
+        FPoint2D pv(point-lineStart);
+
+        //Get dot product (project pv onto normalized direction)
+        float pvdot = dot(dpoint,pv);
+       
+        //Scale line direction vector
+        FPoint2D ds(pvdot * dpoint.x, pvdot * dpoint.y);
+
+        //Subtract this from pointv
+        FPoint2D ap(pv.x - ds.x, pv.y - ds.y);
```

Comment:
We have realization of projection point to line segment in LinearAlgebra

Reply:
Yes, i see, good idea, I'll fix it now